warehouse/clickhouse/clickhouse.go

```diff
@@ -0,0 +1,663 @@
+package clickhouse
+
+import (
+	"compress/gzip"
+	"database/sql"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	_ "github.com/ClickHouse/clickhouse-go"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/rruntime"
+	"github.com/rudderlabs/rudder-server/services/filemanager"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+var (
+	stagingTablePrefix      string
+	maxParallelLoads        int
+	clickhouseDebugLogs     string
+	clickhouseLoadBatchSize string
+	clickhouseQueryPoolSize string
+)
+
+const (
+	host     = "host"
+	dbName   = "database"
+	user     = "user"
+	password = "password"
+	port     = "port"
+	sslMode  = "sslMode"
+)
+
+// clickhouse doesnt support bool, they recommend to use Uint8 and set 1,0
+
+var rudderDataTypesMapToClickHouse = map[string]string{
+	"int":      "Int64",
+	"float":    "Float64",
+	"string":   "String",
+	"datetime": "DateTime",
+	"boolean":  "UInt8",
+}
+
+//TODO: add addition clickhouse types which maps to rudder transformer data types
+var clickhouseDataTypesMapToRudder = map[string]string{
+	"Int64":              "int",
+	"Float64":            "float",
+	"String":             "string",
+	"DateTime":           "datetime",
+	"UInt8":              "boolean",
+	"Nullable(Int64)":    "int",
```

Comment:
add more of these ? eg. add Nullable(Int16) => int also, since if they create a column of int16 manually we will able to map to out int type